modelling/src/main/java/org/axonframework/modelling/entity/child/ListEntityChildModel.java

```diff
@@ -0,0 +1,255 @@
+        private ChildEntityFieldDefinition<P, List<C>> childEntityFieldDefinition;
+        private ChildEntityMatcher<C, CommandMessage<?>> commandTargetMatcher = (child, command) -> true;
+        private ChildEntityMatcher<C, EventMessage<?>> eventTargetMatcher = (child, event) -> true;
+
+        @SuppressWarnings("unused") // Is used for generics
+        private Builder(Class<P> parentClass, EntityModel<C> childEntityModel) {
+            this.childEntityModel = childEntityModel;
+        }
+
+        /**
+         * Sets the {@link ChildEntityFieldDefinition} to use for resolving the child entities from the parent entity
+         * and evolving the parent entity based on the evolved child entities.
+         *
+         * @param fieldDefinition The {@link ChildEntityFieldDefinition} to use for resolving the child entities from
+         *                        the parent entity
+         * @return This builder instance.
+         */
+        public Builder<C, P> childEntityFieldDefinition(
+                @Nonnull ChildEntityFieldDefinition<P, List<C>> fieldDefinition) {
+            this.childEntityFieldDefinition = Objects.requireNonNull(fieldDefinition,
+                                                                     "childEntityFieldDefinition may not be null");
+            return this;
+        }
+
+        /**
+         * Sets the {@link ChildEntityMatcher} to use for matching the child entities to the command. This is used to
+         * filter the child entities based on the command. One entity must match an incoming command. If no match is
+         * found, an exception is thrown. If multiple matches are found, an exception is thrown.
+         *
+         * @param commandTargetMatcher The {@link ChildEntityMatcher} to use for matching the child entities to the
+         *                             command.
+         * @return This builder instance.
+         */
+        public Builder<C, P> commandTargetMatcher(
+                @Nonnull ChildEntityMatcher<C, CommandMessage<?>> commandTargetMatcher) {
+            this.commandTargetMatcher = Objects.requireNonNull(commandTargetMatcher,
+                                                               "commandTargetMatcher may not be null");
+            return this;
+        }
+
+        /**
+         * Sets the {@link ChildEntityMatcher} to use for matching the child entities to the event. This is used to
+         * filter the child entities based on the event. The amount of entities that match an incoming event is not
+         * restricted. Entities that match are evolved, while entities that do not match are ignored.
+         *
+         * @param eventTargetMatcher The {@link ChildEntityMatcher} to use for matching the child entities to the
+         *                           event.
+         * @return This builder instance.
+         */
+        public Builder<C, P> eventTargetMatcher(@Nonnull ChildEntityMatcher<C, EventMessage<?>> eventTargetMatcher) {
```

Comment:
This essentially replaces the `ForwardingModes`, right? Can we adjust those modes into a `ChildEntityMatcher`?

Reply:
Yes, it's similar. The main difference is that the ChildEntityMatcher is a predicate, while the ForwardingMode returns a list of candidates. It is my intention to put all current ForwardingModes into ChildEntityMatcher, but I haven't done that yet. Will pick it up in this PR then.

Reply:
Feel free to pick it up in another PR if that works out better! :pray: 
And, if you haven't done this already :sweat_smile: